Makefile.PL

```diff
@@ -0,0 +1,105 @@
+
+my %WriteMakefileArgs = (
+    NAME                => 'Text::Glob',
+    VERSION_FROM        => 'lib/Text/Glob.pm',
+    ABSTRACT_FROM       => 'lib/Text/Glob.pm',
+    AUTHOR              => 'Richard Clamp <richardc@unixbeard.net>',
+    LICENSE             => 'perl_5',
+
+    META_MERGE => {
+        'meta-spec' => { version => 2 },
+        dynamic_config => 0,
+        resources => {
+            repository => {
+                url => 'https://github.com/richardc/perl-text-glob.git',
+                web => 'https://github.com/richardc/perl-text-glob',
+                type => 'git',
+            },
+            bugtracker => {
+                mailto => 'bug-Text-Glob@rt.cpan.org',
+                web => 'https://rt.cpan.org/Public/Dist/Display.html?Name=Text-Glob',
+            },
+        },
+        x_authority => 'cpan:RCLAMP',
+        x_contributors => [ # manually added, from git shortlog -e -s -n
+             'Tom Hukins <tom@eborcom.com>',
+             'Karen Etheridge <ether@cpan.org>',
+        ],
+    },
+
+    META_ADD => {
+        prereqs => {
+            configure => {
+                requires => {
+                    'ExtUtils::MakeMaker' => '0',
+                },
+            },
+            runtime => {
+                requires => {
+                    'Exporter' => '0',
+                    'constant' => '0',
+                },
+            },
+            test => {
+                requires => {
+                    'Test::More' => '0',
+                },
+            },
+        },
+    },
+);
+
+my $eumm_version  = eval $ExtUtils::MakeMaker::VERSION;
+
+for (qw(configure build test runtime)) {
+    my $key = $_ eq 'runtime' ? 'PREREQ_PM' : uc $_.'_REQUIRES';
+    next unless exists $WriteMakefileArgs{META_ADD}{prereqs}{$_}
+             or exists $WriteMakefileArgs{$key};
+    my $r = $WriteMakefileArgs{$key} = {
+        %{$WriteMakefileArgs{META_ADD}{prereqs}{$_}{requires} || {}},
+        %{delete $WriteMakefileArgs{$key} || {}},
+    };
+    defined $r->{$_} or delete $r->{$_} for keys %$r;
+}
+
+# dynamic prereqs get added here.
+
+$WriteMakefileArgs{MIN_PERL_VERSION} = delete $WriteMakefileArgs{PREREQ_PM}{perl} || 0;
+
+die 'attention developer: you need to do a sane meta merge here!'
+    if keys %{$WriteMakefileArgs{BUILD_REQUIRES}};
+
+$WriteMakefileArgs{BUILD_REQUIRES} = {
+    %{$WriteMakefileArgs{BUILD_REQUIRES} || {}},
+    %{delete $WriteMakefileArgs{TEST_REQUIRES}}
+} if $eumm_version < 6.63_03;
+
+$WriteMakefileArgs{PREREQ_PM} = {
+    %{$WriteMakefileArgs{PREREQ_PM}},
+    %{delete $WriteMakefileArgs{BUILD_REQUIRES}}
+} if $eumm_version < 6.55_01;
+
+delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
+    if $eumm_version < 6.51_03;
+
+delete $WriteMakefileArgs{MIN_PERL_VERSION}
+    if $eumm_version < 6.48;
+
+delete @WriteMakefileArgs{qw(META_ADD META_MERGE)}
+    if $eumm_version < 6.46;
+
+delete $WriteMakefileArgs{LICENSE}
+    if $eumm_version < 6.31;
+
```

Comment:
Why all this stuff gated on version?  Do earlier version of MakeMaker reject these parameters?  Will later versions require further updates?  Why are we replacing 12 lines of declarative configuration with a 105 line monster?